packages/metro-file-map/src/HasteFS.js

```diff
@@ -85,6 +85,53 @@ export default class HasteFS {
     return files;
   }
 
+  /**
+   * Given a search context, return a list of file paths matching the query.
+   * The query matches against normalized paths which start with `./`,
+   * for example: `a/b.js` -> `./a/b.js`
+   */
+  matchFilesWithContext(
+    root: Path,
+    context: $ReadOnly<{
+      /* Should search for files recursively. */
+      recursive: boolean,
+      /* Filter relative paths against a pattern. */
+      filter: RegExp,
+    }>,
+  ): Array<Path> {
+    const files = [];
+    const prefix = './';
+
+    for (const file of this.getAbsoluteFileIterator()) {
+      const filePath = fastPath.relative(root, file);
+
+      const isRelative =
+        filePath && !filePath.startsWith('..') && !path.isAbsolute(filePath);
```

Comment:
isAbsolute check appears to be extraneous now.

Reply:
> I'm not sure normalization is required for the the `..` check to work as this paradigm is cross-platform.

If it's not normalised it might not appear at the start of the path - eg `foo/../../bar` is valid but normalises to `../bar`

Reply:
> isAbsolute check appears to be extraneous now.

I think it's still necessary for Windows tbh, looks like we get absolute paths back when there's no valid relative path between locations:

```
node -e "console.log(path.win32.relative('C:/foo', 'D:/bar'))";
D:/bar
```